### ISSUE TITLE
Does pangeo give an rstudio option?

### ISSUE THREAD

**Issue by MathewBiddle** [ioos/ioos_code_lab#59]
I've been tinkering [1] with deploying Rstudio through mybinder [2] and it's pretty neat. Does pangeo offer the same functionality? Might be a link we could add to the readme.

I've successfully got it working here [3] but it takes a while to build and deploy (~20 mins). 

[1] https://github.com/ioos/bio_mobilization_workshop
[2] https://github.com/binder-examples/r-conda
[3] https://mybinder.org/v2/gh/ioos/bio_mobilization_workshop/main?urlpath=rstudio

**Comment by emiliom** [comment]
I think it depends on what you mean by "pangeo" .... Are you referring to a specific pangeo instance, to the pangeo mybinder, or more broadly the pangeo ecosystem?

I can't imagine that RStudio would ever be deployed quickly via mybinder, unless the environment is already cached. I'm not aware of an active Pangeo instance that has RStudio (Pangeo tends to be Python centric). But as you'll remember, we do have RStudio on the OceanHackWeek JupyterHub, and it worked well AFAIK. See https://ohw.pilot.2i2c.cloud and https://github.com/oceanhackweek/jupyter-image. That's not mybinder, though; it's a customized JupyterHub deployment hosted by 2i2c.org.

**Comment by ocefpaf** [comment]
> I've successfully got it working here [3] but it takes a while to build and deploy (~20 mins).

Yep. You would need to provide your own image to reduce that a bit. What it is doing under the hood is:

- they are probably pulling an image with RStudio, like rocker. It is hard to follow binder's provenance due to the gigantic number of repos and constant changes to be sure;
- then they build your env on top of that.

As far as I know pangeo does not provide an environment with RStudio. You would need to row your own. I did that for the old notebooks_demos. Need to port that here. My delay is b/c a little bird told me RStudio, the company, will be publishing a conda package for it soon. That would make this effort much easier and simpler. Adding a,

```yaml
- rstudio
```

in our yaml files would suffice.

TL;DR stand by for some updates on that front soon.

**Comment by MathewBiddle** [comment]
Has anyone tried it yet? https://anaconda.org/r/rstudio

**Comment by ocefpaf** [comment]
Those are 2 year old and probably broken if installed along side newer packages. We do have a new version in conda-forge [1] but I did not test that yet on binder-like applciations.

[1] https://anaconda.org/conda-forge/rstudio-desktop

**Comment by MathewBiddle** [comment]
This was neat to tinker with, but not something I'm concerned with. If it comes up again we can reopen, or start something new. For now its not a feature folks are clambering for.
